### PR TITLE
Update as_ikea.py

### PR DESCRIPTION
**Pull request by vladlemberg**
- added more ikea letters with random selection. -)
- proper capitalization.

### CODE REVIEW

File: as_ikea.py [hunk 1]
```diff
@@ -1,16 +1,50 @@
...
+
+    for index, letter in enumerate(name.lower(), start=0):
+        if mapper.get(letter):
+            name = replacer(name, random.choice(mapper[letter]), index)
+
+    return name[::-1].title()
+
+
+def replacer(s: str, newstring: str, index: int, nofail: bool = False) -> str:
+    """
```
Comment (epogrebnyak) [inline as_ikea.py:30]: Is intent to replace just one occurrence of the letter here? 
I would eliminate a flag here for alternative behavior, if we wanted branching perhaps two functions is beyter

File: as_ikea.py [hunk 2]
```diff
@@ -1,16 +1,50 @@
...
+    # raise an error if index is outside of the string
+    if not nofail and index not in range(len(s)):
+        raise ValueError("index outside given string")
+
+    # insert the new string between "slices" of the original
+    return s[:index] + newstring + s[index + 1:]
+
 
-assert as_ikea("Ivanov") == 'Vönävi' #fails - add .lower() and .capitalize()
+#assert as_ikea("Ivanov") == 'Vönävi'
```
Comment (epogrebnyak) [inline as_ikea.py:43]: I will need to restore some tests either as asserts, or as real unit tests.

**Review by epogrebnyak — COMMENTED** [review]
Thanks for contributing more letters and functionality! I was not expecting any inputs on the post and glad it arrived. I would have done several things slightly differently, but I can amend them mysefl after merging.

**Comment by epogrebnyak** [comment]
Should I merge as is?

File: as_ikea.py [hunk 3]
```diff
@@ -1,16 +1,50 @@
...
+    Convert letters to ikea letters
+    :param: string
+    :return: ikea string
+    """
+    mapper = dict(a=["à", "á", "â", "ä", "æ", "ã", "ā"],
+                  o=["ô", "ö", "ò", "ó", "œ", "ø", "ō"],
+                  u=["û", "ü", "ù", "ú", "ū"],
+                  e=["è", "é", "ê", "ë", "ē", "ė"])
+
+    for index, letter in enumerate(name.lower(), start=0):
```
[the remark this answers is not in the document]
Comment (vladlemberg) [reply as_ikea.py:22]: it does not. can be omitted.

**Comment by vladlemberg** [comment]
fun stuff -)

**Comment by epogrebnyak** [comment]
I may end up learning Swedish or becoming an IKEA brand ambassador)

чт, 29 окт. 2020 г., 23:47 Vladimir Lemberg <notifications@github.com>:

**Comment by epogrebnyak** [comment]
Thanks for supporting the idea and hope you enjoyed it as I do

There some more serious depths on how to make really Swedish-sounding names
with some kind of a check against a corpus of words, but not sure if can
make it to there.

чт, 29 окт. 2020 г., 23:47 Vladimir Lemberg <notifications@github.com>:

**Pull request merged**